src/elf.rs

```diff
@@ -148,6 +149,12 @@ where
     fn is_little_endian(&self) -> bool {
         self.elf.little_endian
     }
+
+    fn has_debug_symbols(&self) -> bool {
+        self.section_data_by_name(".debug_info").is_some()
```

Comment:
I'm not confident that this is correct. What if the file was generated with `.debug_line` only? What's the intended use for this? Should calllers only look for `.gnu_debuglink` if this returns false? I think it might be better to always use `.gnu_debuglink` if it exists.

Reply:
Actually, `.debug_info` exists even with `-C debuginfo=1`. Still might be better to always check for `.gnu_debuglink` though.

Reply:
I'm not 100% sure, honestly, except that `.debug_info` is a pretty strong signal. Calling `strip` on a binary will definitely remove `.debug_info` by default. This is mostly intended as a check for "can I load debug info from this binary, or do I need to look for an external symbol file?"

Reply:
> This is mostly intended as a check for "can I load debug info from this binary, or do I need to look for an external symbol file?"

There's also gdb "mini debug info" to consider: https://sourceware.org/gdb/onlinedocs/gdb/MiniDebugInfo.html#MiniDebugInfo

With this an executable might contain some minimal info (line table) but more data might be available elsewhere.  gdb looks for direct debug info first; then tries separate debuginfo; then falls back to minidebug.

Reply:
Following whatever gdb does here is probably the best bet for compatibility's sake.

Reply:
This looks fine then.